app/models/product_user.rb

```diff
@@ -2,9 +2,14 @@ class ProductUser < ActiveRecord::Base
   belongs_to :user
   belongs_to :product
   belongs_to :product_access_group
-  
-  validates_numericality_of :product_id, :user_id, :approved_by, :only_integer => true
-  validates_uniqueness_of :user_id, :scope => :product_id, :message => 'is already approved'
 
-  before_create lambda { |pu| pu.approved_at = Time.zone.now }
-end
\ No newline at end of file
+  validates_numericality_of :product_id, :user_id, :approved_by, only_integer: true
+  validates_uniqueness_of :user_id, scope: :product_id, message: "is already approved"
+
+  before_create -> (product_user) { product_user.approved_at = Time.zone.now }
+
+  def training_request
```

Comment:
Maybe I just haven't seen it yet, but where does this get used? And since we're destroying the training request once the user is approved, wouldn't this always be empty?